README.md

```diff
@@ -5,76 +5,143 @@
+Additionally, you need to specify an event listener. HawkTracer currently provides 2 listeners:
+* TCP listener, which streams events over the network
+* File dump listener, which saves events to a file.
+
+Moreover, HawkTracer allows to provide user-defined listeners.
+Code below presents example HawkTracer (un)initialization:
+```cpp
+int main(int argc, char** argv)
+{
+  ht_init(argc, argv); // initialize library
+  HT_Timeline* timeline = ht_global_timeline_get(); // timeline, where all events are posted. You can define your own timeline, or use global timeline
+  HT_FileDumpListener listener;
+  const size_t buffer_size = 4096; // size of internal listener's buffer
+  ht_file_dump_listener_init(&listener, "file_name.htdump", buffer_size); // initialize listener
+  ht_timeline_register_listener(timeline, ht_file_dump_listener_callback, &listener); // register listener to a timeline
+
+  // your code goes here...
+  
+  ht_timeline_flush(timeline); // flush all remaining events from timeline
+  ht_timeline_unregister_all_listeners(timeline); // unregister listeners from timeline
+  ht_file_dump_listener_deinit(&listener); // deinitialize listener
+  ht_deinit(); // uninitialize library
+
+  return 0;
+}
 
 ```
---@TracepointStart Identifier
+The code registers file dump listener, which saves all the events to a file `file_name.htdump`. The file should be then converted to a viewer's format ([see here for details](#collect-the-data)).
 
--- Lua code here
+#### Instrumenting the code
+HawkTracer requires explicit code instrumentation. The library provides a few helper macros for reporting data to a timeline:
+```cpp
+// Pushes any type of event to a timeline
+HT_TIMELINE_PUSH_EVENT(TIMELINE, EVENT_TYPE, EVENT_PARAMETERS,...)
 
---@TracepointStop
+// Reports a duration of specific block of code (available only for C++ or C GNU compiler)
+HT_TP_STRACEPOINT(TIMELINE, LABEL)
 
+// The same as above, but automatically sets label to current function name
+HT_TP_FUNCTION(TIMELINE)
+```
+There are few macros which report events to a global timeline, they're prefixed with `G_`:
+```cpp
+HT_TP_G_STRACEPOINT(LABEL)
+HT_TP_G_FUNCTION()
+```
+For example, you can instrument following code:
+```cpp
+void foo()
+{
+  HT_TP_G_FUNCTION() // measure duration of foo function execution
+  
+  for (int i = 0; i < 100; i++)
+  {
+    HT_TP_G_STRACEPOINT("in-loop") // measure duration of single loop iteration
+	bar();
+  }
+}
 ```
-
-### Add to your project's build system
-
-[TBD]
 
 ### Collect the data
 
-[TBD]
+For now HawkTracer provides a simple application for converting event stream to a JSON format which can be consumed by the viewing application: `hawktracer-to-json`. You can use it with a file or with a network stream. Assuming your events have been saved to `file_name.htdump` file, you can generate the JSON file running following command:
+```bash
+hawktracer-to-json --source file_name.htdump --output output_file.json
```

Comment:
[Tiny] It would be nice to add an example with network stream.

Reply:
Yeah, do agree. However, I didn't want to create this readme too long. When we have more detailed documentation, we should definitely add a link here to a page where network listeners/streams are described.